### DISCUSSION TITLE
OpenSeadragon Course

### DISCUSSION THREAD

**Discussion by iangilman** [openseadragon/openseadragon#1990] [General]
I'm in the beginning stages of working on an OpenSeadragon learning course to be published on [Newline](https://www.newline.co/). I intend for it to cover the basics (preparing your images and setting up a viewer, as well as some background history and theory), and more advanced topics, like:

* Working with the coordinate system
* Adding and manipulating overlays
* Custom animations
* Multi-image scenarios

What do you think? Is this something you would be interested in? What topics would you like to see covered? Hit me with ideas! :-)

**Comment by mmt1821** [comment]
Dear Ian,

what a great idea for the course.

OpenSeaDragon is really nice framework to build scalable visualisations (of images and more), like the one you have done with flickr.
Personally, I would like to see how to use OSD to browse a complex structure of objects, some of which are images, or they they have image/iconic representations.

I would also like to see how to use OSD to implement infinite canvas abstraction, and have a smooth animated way to navigate from scene to scene. 

Bird's eye view of the OSD world (actually multiple views of the same object structure), would be nice to see.


Thanks for contributing to OSD.

--manolis

**Reply by iangilman** [reply]
Awesome… Thank you for the ideas! 

Can I ask for a little more detail? When you say infinite canvas navigating from scene to scene, are you meaning like a comic book viewer, like http://www.driftory.com/? And what do you mean by "Bird's eye view of the OSD world (actually multiple views of the same object structure)"? Like a bird's eye view of the inner workings of OSD? Or using OSD to get a birds eye view of some data?

Thank you again :-)

**Comment by AndrasSCH** [comment]
Hi Ian, 
Thank you for thinking of a course- great idea! I would be very interested.  Do you have details, like when, fee, length and platform? I am building a site for pathology education.  OSD is now connected with CMS whole slide image database. I'd like to expand the functionality with zoom to interest and svg annotations.  Really look forward to this and thank you for being so activity in helping the community. Andras

**Reply by iangilman** [reply]
I'm glad you're interested! I definitely plan to cover zooming to points of interest and adding SVG annotations. Unfortunately, progress has been slow as I've been busy with other things. That said, I do have a start on it. Would you be interested in being a beta reader? Basically someone to take an early look and give me feedback on it. 

The platform is https://www.newline.co/. I'm not sure about the fee or length yet.

**Reply by AndrasSCH** [reply]
Hi Ian, 
Sounds great. Yes I’d love to be a beta tester. I am not a software engineer but know some basic html, css and js. Please let me know how much time commitment it will take. My email is schaffer.andras@gmail.com

**Reply by iangilman** [reply]
Lovely! I'll email you when I'm ready to add you. Regarding time commitment, ideally you would read through the course and follow along coding-wise, making notes of what was confusing (and what you liked!). A few hours?

**Reply by AndrasSCH** [reply]
Perfect. I have Thursdays and Fridays available, weekends if needed, and I am off with family 8/5-8/13.  Let me know. Looking forward to this.

**Comment by iangilman** [comment]
Wow, has it really been four years since I started the preparing the course??

Well, it's finally released! I've put a lot of time and care into it. It's meant to be the best introduction to working with OpenSeadragon, from basic concepts to preparing images to coordinate systems, animation, and beyond. If you're new to OpenSeadragon, this is a great way to get started! If you've already been around the block, you can still deepen your knowledge. If you're already plenty deep, I bet you know someone who could use it... please help spread the word! ❤️ 

It's 10% off until June 9 with this link: https://www.newline.co/courses/openseadragon-deep-dive?code=OpenSeadragonMastery

**Comment by iangilman** [comment]
![course2](https://github.com/user-attachments/assets/8242717a-b25c-4117-af8f-31cb9b600d18)

As you can see, I'm in the process of trying to get the word out! I'd love to hear if you have any thoughts for what venues to share it with. Are there communities of archivists who might be interested? Medical imaging folks? Astronomers? Etc.

**Reply by hrxn** [reply]
Also, museums, gallerists, professional photographers etc.

**Reply by iangilman** [reply]
Absolutely! 😊

**Comment by AndrasSCH** [comment]
Hi Ian, please post on https://digitalpathologyassociation.org/
Great stuff and and thanks for the course!
Andras

On Mon, Jun 9, 2025 at 9:21 AM Ian Gilman ***@***.***> wrote:

> Absolutely! 😊
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openseadragon/openseadragon/discussions/1990#discussioncomment-13411835>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AR5JVQCSQZ4YGMDF5RXU5FL3CWYBDAVCNFSM6AAAAAB6RPONP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTGNBRGE4DGNI>
> .
> You are receiving this because you commented.Message ID:
> <openseadragon/openseadragon/repo-discussions/1990/comments/13411835@
> github.com>
>


-- 
*Andras Schaffer, M.D., Ph.D.*
*Associate Professor*
*Associate Program Director, Dermatology Residency Training Program*

*Director, Dermatopathology Education*
Division of Dermatology
University of Central Florida College of Medicine
Dermatology Associates of Tallahassee
215.518.7823 cell
850.878.1261 office

**Reply by iangilman** [reply]
@AndrasSCH Thank you for the lead!

Looks like I would have to join as a member to be able to post? The sign-up form looks daunting, and I wonder if I would be welcome if I signed up just to promote my course.

I assume you're a member already? Might it be possible for you to post about the course?

**Comment by AndrasSCH** [comment]
Hi Ian,
I’ll be happy to promote the course!
I’ll look into this this weekend.
Andras

*Andras Schaffer, M.D., Ph.D.*
*Associate Professor*
*Associate Program Director, Dermatology Residency Training Program*

*Director, Dermatopathology Education*
Division of Dermatology
University of Central Florida College of Medicine
Dermatology Associates of Tallahassee
215.518.7823 cell
850.878.1261 office

**Reply by iangilman** [reply]
@AndrasSCH Wonderful, thank you! Let me know if you need anything for me 😊

**Comment by AndrasSCH** [comment]
Ian, I posted it on LinkedIn and tagged the digital pathology community.
Hope you are well. Happy 4th.

On Sat, Jul 5, 2025 at 4:08 AM Ian Gilman ***@***.***> wrote:

> @AndrasSCH <https://github.com/AndrasSCH> Wonderful, thank you! Let me
> know if you need anything for me 😊
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openseadragon/openseadragon/discussions/1990#discussioncomment-13665474>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AR5JVQCSVFOVF33NBXWMRQL3G27IBAVCNFSM6AAAAAB6RPONP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTGNRWGU2DONA>
> .
> You are receiving this because you were mentioned.Message ID:
> <openseadragon/openseadragon/repo-discussions/1990/comments/13665474@
> github.com>
>


-- 
*Andras Schaffer, M.D., Ph.D.*
*Associate Professor*
*Associate Program Director, Dermatology Residency Training Program*

*Director, Dermatopathology Education*
Division of Dermatology
University of Central Florida College of Medicine
Dermatology Associates of Tallahassee
215.518.7823 cell
850.878.1261 office
